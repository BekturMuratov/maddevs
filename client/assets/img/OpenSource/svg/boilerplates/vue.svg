<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>