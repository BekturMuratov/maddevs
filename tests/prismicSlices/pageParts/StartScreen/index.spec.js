import { render, screen } from '@testing-library/vue'
import { shallowMount } from '@vue/test-utils'
import StartScreenMain from '@/prismicSlices/pageParts/StartScreen'
import { getFakePropsByParams } from './testsUtils'

const backgrounds = {
  white: '#ffffff',
  black: '#111213',
  grey: '#f5f7f9',
}

const apiData = {
  background: 'black',
  animation: 'fade-up',
}

describe('StartScreen slice', () => {
  it('should correctly render CardGrid component', () => {
    const props = getFakePropsByParams(apiData)
    props.slice.variation = 'default-slice'
    const { container } = render(StartScreenMain, {
      props,
    })

    expect(screen.queryByTestId('start-screen-variation')).not.toBeNull()
    expect(screen.queryByTestId('description-variation')).toBeNull()
    expect(container).toMatchSnapshot()
  })

  it('should correctly render CardGridWithIcon component', () => {
    const props = getFakePropsByParams(apiData)
    props.slice.variation = 'cardGridWithIcon'
    const { container } = render(StartScreenMain, {
      props,
    })

    expect(screen.queryByTestId('start-screen-variation')).toBeNull()
    expect(screen.queryByTestId('description-variation')).not.toBeNull()
    expect(container).toMatchSnapshot()
  })

  describe('sliceBackground computed method', () => {
    it('should return white hex', () => {
      const wrapper = shallowMount(StartScreenMain, {
        propsData: getFakePropsByParams({
          ...apiData,
          background: 'white',
        }),
      })

      expect(wrapper.vm.sliceBackground).toBe(backgrounds.white)
    })

    it('should return grey hex', () => {
      const wrapper = shallowMount(StartScreenMain, {
        propsData: getFakePropsByParams({
          ...apiData,
          background: 'grey',
        }),
      })

      expect(wrapper.vm.sliceBackground).toBe(backgrounds.grey)
    })

    it('should return black hex', () => {
      const wrapper = shallowMount(StartScreenMain, {
        propsData: getFakePropsByParams({
          ...apiData,
          background: 'black',
        }),
      })

      expect(wrapper.vm.sliceBackground).toBe(backgrounds.black)
    })

    it('should return null', () => {
      const wrapper = shallowMount(StartScreenMain, {
        propsData: getFakePropsByParams({
          ...apiData,
          background: 'unknown',
        }),
      })

      expect(wrapper.vm.sliceBackground).toBeNull()
    })
  })
})
