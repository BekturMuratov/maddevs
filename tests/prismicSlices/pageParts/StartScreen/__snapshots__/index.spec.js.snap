// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartScreen slice should correctly render CardGrid component 1`] = `
<div>
  <section
    class="card-grid-main"
    style="background-color: rgb(17, 18, 19);"
  >
    <section
      class="start-screen-slice"
      data-testid="start-screen-variation"
      id="transparent-header-area"
      primary="[object Object]"
      variation="default-slice"
    >
      <img
        alt="Image"
        class="start-screen-slice__image"
        data-testid="start-screen-slice__image"
        height="969"
        style="background-color: rgb(17, 18, 19);"
        width="1680"
      />
       
      <div
        class="container"
      >
        <div
          class="start-screen-slice__content"
          style="opacity: 1;"
        >
          <h1
            class="start-screen-slice__title"
            data-testid="start-screen-slice__title"
          />
           
          <!---->
        </div>
      </div>
    </section>
     
    <!---->
  </section>
</div>
`;

exports[`StartScreen slice should correctly render CardGridWithIcon component 1`] = `
<div>
  <section
    class="card-grid-main"
    style="background-color: rgb(17, 18, 19);"
  >
    <!---->
     
    <section
      class="description-slice"
      data-testid="description-variation"
      primary="[object Object]"
      variation="cardGridWithIcon"
    >
      <div
        class="container"
      >
        <div
          class="description-slice__content"
        >
          <h1
            class="description-slice__title"
            data-testid="description-slice__title"
          />
           
          <!---->
        </div>
      </div>
    </section>
  </section>
</div>
`;
